src/Controls/samples/Controls.Sample/MyApp.cs

```diff
@@ -1,21 +1,26 @@
 using System;
-using Maui.Controls.Sample.Services;
+using MauiSampleApp.Services;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Maui;
 using Microsoft.Maui.Controls.Compatibility;
 
-namespace Maui.Controls.Sample
+namespace MauiSampleApp
 {
-	public class MyApp : MauiApp
+	public class MyApp : IApplication
 	{
-		public MyApp(ITextService textService)
+		public MyApp(IServiceProvider services, ITextService textService)
 		{
+			Services = services;
+
 			Console.WriteLine($"The injected text service had a message: '{textService.GetText()}'");
 		}
 
-		public override IWindow CreateWindow(IActivationState state)
+		public IServiceProvider Services { get; }
+
+		public IWindow CreateWindow(IActivationState activationState)
 		{
-			Forms.Init(state);
+			Forms.Init(activationState);
```

Comment:
Will this remain in the dev's code? I thought the HostBuilder took care of all this and we wouldn't see "Forms" anymore.

Reply:
I imagine once we get all the code inside `Init` reorganized into better places this will go away. This is mainly here to make the shims and old code happy for the time being.

Though we'll probably want to give some guidance where users put 3rd party Init's for the time being

Reply:
This is also something that we can do if we have a way to register delegates for a pre build and post build run. Then also a delegate to run each activity. Users may want to do things when those events happen. This is all very likely covered by the lifecycle things already that @jsuarezruiz is doing in the #495 PR. We can hook into the various events and do this for the users. And then that can be the recommendation for library authors or the user can do manual init in there for old libs.